File: library/src/main/java/com/sw926/imagefileselector/CompressFormatUtils.java
package com.sw926.imagefileselector;

import android.graphics.Bitmap;
import android.os.Build;

/**
 * Created by sunwei on 15/11/16.
 */
public class CompressFormatUtils {

    public static Bitmap.CompressFormat parseFormat(String fileName) {

        int dotPos = fileName.lastIndexOf(".");
        if (dotPos <= 0) {
            return Bitmap.CompressFormat.JPEG;
        }
        String ext = fileName.substring(dotPos + 1);
        if (ext.equalsIgnoreCase("jpg") || ext.equalsIgnoreCase("jpeg")) {
            return Bitmap.CompressFormat.JPEG;
        }
        if (ext.equalsIgnoreCase("png")) {
            return Bitmap.CompressFormat.PNG;
        }
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.ICE_CREAM_SANDWICH) {
            if (ext.equalsIgnoreCase("webp")) {
                return Bitmap.CompressFormat.WEBP;
            }
        }
        return Bitmap.CompressFormat.JPEG;
    }

    public static String getExt(Bitmap.CompressFormat format) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.ICE_CREAM_SANDWICH) {
            if (format == Bitmap.CompressFormat.WEBP) {
                return ".webp";
            }
        }

        if (format == Bitmap.CompressFormat.PNG) {
            return ".png";
        }
        return ".jpg";
    }

    public static String getExt(String fileName) {
        Bitmap.CompressFormat format = parseFormat(fileName);
        return getExt(format);
    }
}

File: library/src/main/java/com/sw926/imagefileselector/CompressFormatUtils.kt
package com.sw926.imagefileselector

import android.graphics.Bitmap.CompressFormat

/**
 * Created by sunwei on 15/11/16.
 */
object CompressFormatUtils {
    fun parseFormat(fileName: String): CompressFormat {
        val dotPos = fileName.lastIndexOf(".")
        if (dotPos <= 0) {
            return CompressFormat.JPEG
        }
        val ext = fileName.substring(dotPos + 1)
        if (ext.equals("jpg", ignoreCase = true) || ext.equals("jpeg", ignoreCase = true)) {
            return CompressFormat.JPEG
        }
        return if (ext.equals("png", ignoreCase = true)) {
            CompressFormat.PNG
        } else CompressFormat.JPEG
    }

    fun getExt(format: CompressFormat): String {
        if (format == CompressFormat.WEBP) {
            return ".webp"
        }
        return if (format == CompressFormat.PNG) {
            ".png"
        } else ".jpg"
    }

    fun getExt(fileName: String): String {
        val format = parseFormat(fileName)
        return getExt(format)
    }
}